src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/IntrospectionController.cs

```diff
@@ -0,0 +1,56 @@
+using System.Threading.Tasks;
+
+using Altinn.Platform.Authentication.Model;
+using Altinn.Platform.Authentication.Services.Interfaces;
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.Platform.Authentication.Controllers
+{
+    /// <summary>
+    /// Controller for introspection requests to Platform Authentication
+    /// </summary>
+    [Route("authentication/api/v1/introspection")]
+    [ApiController]
+    public class IntrospectionController : ControllerBase
+    {
+        private readonly IEFormidlingAccessValidator _eFormidlingAccessValidator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IntrospectionController"/> class.
+        /// </summary>
+        /// <param name="eFormidlingAccessValidator">The eFormidling access validator service</param>
+        public IntrospectionController(IEFormidlingAccessValidator eFormidlingAccessValidator)
+        {
+            _eFormidlingAccessValidator = eFormidlingAccessValidator;
+        }
+
+        /// <summary>
+        /// Validate token.
+        /// </summary>
+        [AllowAnonymous]
+        [HttpPost]
+        [Produces("application/json")]
+        [Consumes("application/x-www-form-urlencoded")]
+        [ProducesResponseType(typeof(IntrospectionResponse), StatusCodes.Status200OK)]
+        public async Task<ActionResult<IntrospectionResponse>> ValidateToken([FromForm] IntrospectionRequest request)
+        {
+            string token = request.Token;
+
+            switch (request.TokenTypeHint)
```

Comment:
While I understand the idea behind the switch it is currently unnecessary. The outcome is the same for all branches.

Reply:
Worried that expansion of tokens we support in the future will result in us not following the standard. Any suggestions on how to make it clear when we only have the once case for now?

Reply:
That's fair, but then should we also make it more obvious that the hint is just a hint? The little bit I read about introspection enpoints said something about trying other "validators" if the token turned out invalid when checked based on the hint.

It's also tempting to move the logic to a service. In the name of separating "presentation" logic from "business" logic.

Reply:
created a new service, named it IAuthentication thinking we can extend it with all other logic currently in the controllers, what do you thing, @SandGrainOne  ?